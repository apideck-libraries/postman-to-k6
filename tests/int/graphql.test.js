/* eslint-disable space-before-function-paren */

import convertFile from '../../lib/convert/file';

function render(body) {
  const content = JSON.stringify(body);
  const escaped = content.replace(/\\/g, '\\\\');
  return `'${escaped}'`;
}

test('body simple', async () => {
  const [main] = await convertFile(
    'test/material/2.1/graphql-body-simple.json'
  );
  expect(main).toMatchSnapshot();

  //   const query =
  //     '' +
  //     `{
  //     user {
  //         name
  //     }
  // }`;
  //   const body = { query };
  //   t.is(
  //     main,
  //     `// Auto-generated by the postman-to-k6 converter
  //
  // import "./libs/shim/core.js";
  //
  // export let options = { maxRedirects: 4 };
  //
  // const Request = Symbol.for("request");
  // postman[Symbol.for("initial")]({
  //   options
  // });
  //
  // export default function() {
  //   postman[Request]({
  //     name: "Test Request",
  //     method: "POST",
  //     address: "http://graphql.example.com/",
  //     data: ${render(body)}
  //   });
  // }
  // `
  //   );
});

test('body var', async () => {
  const [main] = await convertFile('test/material/2.1/graphql-body-var.json');
  expect(main).toMatchSnapshot();

  //   const query =
  //     '' +
  //     `query UserName($id: UserId) {
  //     user(id: $id) {
  //         name
  //     }
  // }`;
  //   const variables =
  //     '' +
  //     `{
  // \t"id": "{{userId}}"
  // }`;
  //   const body = { query, variables };
  //   t.is(
  //     main,
  //     `// Auto-generated by the postman-to-k6 converter
  //
  // import "./libs/shim/core.js";
  //
  // export let options = { maxRedirects: 4 };
  //
  // const Request = Symbol.for("request");
  // postman[Symbol.for("initial")]({
  //   options,
  //   collection: {
  //     userId: "5"
  //   }
  // });
  //
  // export default function() {
  //   postman[Request]({
  //     name: "Test Request",
  //     method: "POST",
  //     address: "http://graphql.example.com/",
  //     data:
  //       ${render(body)}
  //   });
  // }
  // `
  //   );
});
