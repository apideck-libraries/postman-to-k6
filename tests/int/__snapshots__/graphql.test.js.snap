// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`body simple 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options
});

export default function() {
  postman[Request]({
    name: "Test Request",
    method: "POST",
    address: "http://graphql.example.com/",
    data: '{"query":"{\\\\n    user {\\\\n        name\\\\n    }\\\\n}"}'
  });
}
"
`;

exports[`body var 1`] = `
"// Auto-generated by the postman-to-k6 converter

import "./libs/shim/core.js";

export let options = { maxRedirects: 4 };

const Request = Symbol.for("request");
postman[Symbol.for("initial")]({
  options,
  collection: {
    userId: "5"
  }
});

export default function() {
  postman[Request]({
    name: "Test Request",
    method: "POST",
    address: "http://graphql.example.com/",
    data:
      '{"query":"query UserName($id: UserId) {\\\\n    user(id: $id) {\\\\n        name\\\\n    }\\\\n}","variables":"{\\\\n\\\\t\\\\"id\\\\": \\\\"{{userId}}\\\\"\\\\n}"}'
  });
}
"
`;
